lib/commands/instrument.js

```diff
@@ -92,40 +97,45 @@ exports.builder = function (yargs) {
     .example('$0 instrument ./lib ./output', 'instrument all .js files in ./lib with coverage and output in ./output')
 }
 
+const errorExit = msg => {
+  console.error(`nyc instrument failed:  ${msg}`)
+  process.exitCode = 1
+}
+
 exports.handler = function (argv) {
-  if (argv.output && (path.resolve(argv.cwd, argv.input) === path.resolve(argv.cwd, argv.output))) {
-    console.error(`nyc instrument failed: cannot instrument files in place, <input> must differ from <output>`)
-    process.exitCode = 1
-    return
+  if (argv.output && !argv.inPlace && (path.resolve(argv.cwd, argv.input) === path.resolve(argv.cwd, argv.output))) {
+    return errorExit(`cannot instrument files in place, <input> must differ from <output>.  Set '--in-place' to force`)
   }
 
   if (path.relative(argv.cwd, path.resolve(argv.cwd, argv.input)).startsWith('..')) {
-    console.error(`nyc instrument failed: cannot instrument files outside of project root directory`)
-    process.exitCode = 1
-    return
+    return errorExit('cannot instrument files outside project root directory')
+  }
+
+  if (argv.delete && argv.inPlace) {
+    return errorExit(`cannot use '--delete' when instrumenting files in place`)
   }
 
   if (argv.delete && argv.output && argv.output.length !== 0) {
     const relPath = path.relative(process.cwd(), path.resolve(argv.output))
     if (relPath !== '' && !relPath.startsWith('..')) {
       rimraf.sync(argv.output)
     } else {
-      console.error(`nyc instrument failed: attempt to delete '${process.cwd()}' or containing directory.`)
-      process.exit(1)
+      return errorExit(`attempt to delete '${process.cwd()}' or containing directory.`)
     }
   }
 
+  console.log(`Running nyc from ${process.cwd()}`)
+
   // If instrument is set to false enable a noop instrumenter.
   argv.instrumenter = (argv.instrument)
     ? './lib/instrumenters/istanbul'
     : './lib/instrumenters/noop'
 
-  const nyc = new NYC(argv)
+  if (argv.inPlace) {
+    argv.output = argv.input
+    argv.completeCopy = false
```

Comment:
Should `--in-place --complete-copy` be an error like `--delete --in-place`?

Reply:
I'm thinking it should be a no-op and shouldn't require a warning.  If you could enable the `--complete-copy` switch for `instrument --in-place` you would expect the same result whether or not it was active.